crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
     }
 }
 
-/// Parse and run command line arguments
-pub fn run() -> sc_cli::Result<()> {
-    let mut cli = Cli::from_args();
-
+fn set_madara_path(cli: &mut Cli) {
     // alias madara_path <> base_path
-    // TODO also alias tmp (tmp generates random base_paths that are not specified within
-    // the command)
-    let madara_path = match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
+    match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
         (Some(madara_path), _) => {
-            cli.run.run_cmd.shared_params.base_path = Some(madara_path.clone());
-            madara_path.to_str().unwrap().to_string()
+            cli.run.run_cmd.shared_params.base_path = Some(madara_path);
         }
         (_, Some(base_path)) => {
-            cli.run.madara_path = Some(base_path.clone());
-            base_path.to_str().unwrap().to_string()
+            cli.run.madara_path = Some(base_path);
         }
         _ => {
             let home_path = std::env::var("HOME").unwrap_or(std::env::var("USERPROFILE").unwrap_or(".".into()));
             let path = format!("{}/.madara", home_path);
             cli.run.run_cmd.shared_params.base_path = Some((path.clone()).into());
-            cli.run.madara_path = Some((path.clone()).into());
-            path
+            cli.run.madara_path = Some(path.into());
```

Comment:
Also this would be better written like this:

```rust
let path = cli.run.madara_path
                       .or(cli.run.run_cmd.shared_params.base_path)
                       .or_else(|| {
           let home_path = std::env::var("HOME").unwrap_or(std::env::var("USERPROFILE").unwrap_or(".".into()));
            format!("{}/.madara", home_path);
});

cli.run.run_cmd.shared_params.base_path = path;
cli.run.madara_path = path
```

Reply:
We have been using that architecture for some time now. We can't merge both arguments. We fully manage the Madara path and we know when and how it is set up. The same doesn't apply to base_path. That argument is managed by substrate and can change during the program's runtime (e.g. when you set the `-tmp`) flag. Wdyt?

Reply:
But if they are always the same. Can't we just get rid of madara_path and just use base_path for everything instead?